models/issue_comment.go

```diff
@@ -485,6 +491,45 @@ func createAssigneeComment(e *xorm.Session, doer *User, repo *Repository, issue
 	})
 }
 
+func createAddedDeadlineComment(e *xorm.Session, doer *User, repo *Repository, issue *Issue, dateUnix util.TimeStamp) (*Comment, error) {
```

Comment:
I think that you could pack all of these `create...DeadlineComment` functions into one function with an additional parameter `type` (please also check if the type is a valid deadline type).

Reply:
Done, good idea!